### PR TITLE
fix: ignore and remove server.pid in docker env

### PR DESCRIPTION
**Pull request by roschaefer**
See: https://docs.docker.com/samples/rails/#define-the-project

fix: #1028

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -e
+
+# Remove a potentially pre-existing server.pid for Rails.
+rm -f /app/tmp/pids/server.pid
+
+# Then exec the container's main process (what's set as CMD in the Dockerfile).
```
Comment (tillprochaska) [inline entrypoint.sh:7]: This is a little inaccurate, as `CMD` only sets a default command that can be overwritten (e.g. via command line parameters or `docker-compose`)…

File: Dockerfile [hunk 2]
```diff
@@ -16,6 +16,10 @@ RUN apk --update add \
     less \
     libsodium
 
+COPY entrypoint.sh /usr/bin/
+RUN chmod +x /usr/bin/entrypoint.sh
```
Comment (tillprochaska) [inline Dockerfile:20]: What’s the reason to apply the execute permission only after copying to the container? Why can’t we just check in the script with the correct permissions in the beginning?

**Review by tillprochaska — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (roschaefer) [reply Dockerfile:20]: Because it shouldn't be executed in a non-virtualized environment like your local machine.

File: entrypoint.sh (hunk 1 above)
Comment (roschaefer) [reply entrypoint.sh:7]: :thinking: err, yes? That's what we want, right?

**Pull request merged**